### PR TITLE
Survival results update

### PR DESCRIPTION
**Pull request by jjc2718**
**PR overview:**

Following #65 , we also wanted to update the survival results to use the new data shuffling approach. Instead, we actually ended up dropping the shuffled models for survival (they tend to have issues with convergence) and deciding just to plot raw concordance index values. As a baseline/"negative control", we're instead using a model that only uses clinical covariates (age, sample mutation burden, sample cancer type).

Overall, our models tend to (slightly) outperform the baseline (grey dotted line), but there's not much difference between -omics types:

image

image

**Code changes:**

* Added survival results QC script: `06_predict_survival/nbconverted/survival_results_qc.py` (this is new)
* Moved `06_predict_survival/nbconverted/plot_survival_pc_comparison.py` to `06_predict_survival/nbconverted/plot_survival_results.py` and made a few changes to generate paper figure versions with clinical-only baselines
* Renamed `06_predict_survival/nbconverted/plot_survival_heatmaps.py` (this has already been reviewed, you don't need to look at it again)
* Update scripts in `06_predict_survival` to run covariate-only baseline
* Add bash scripts to run experiments

### CODE REVIEW

File: 06_predict_survival/nbconverted/plot_survival_results.py [hunk 1]
```diff
@@ -37,523 +39,497 @@
...
-#     
-# As a metric, for now we're just using the [censored concordance index](https://scikit-survival.readthedocs.io/en/latest/api/generated/sksurv.metrics.concordance_index_censored.html). Essentially, this compares the actual order of events (i.e. death or tumor progression) in the test dataset vs. the order of events predicted by the model in the test samples. A higher concordance index = better prediction.
 
-# In[6]:
+# In[4]:
 
 
-sns.set({'figure.figsize': (18, 15)})
-fig, axarr = plt.subplots(3, 1)
+# get baseline predictor results, using non-omics covariates only
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply 06_predict_survival/nbconverted/plot_survival_results.py:79]: Why would you expect only considering clinical metadata to perform worse compared to the other data types?

**Review by ajlee21 — APPROVED** [review]
Interesting result. I have some conceptual questions to try to understand your analysis. It looks like you're performing several comparisons, maybe it would help to define the experiment in the notebook somewhere.

File: 06_predict_survival/nbconverted/survival_results_qc.py [hunk 2]
```diff
@@ -0,0 +1,431 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Survival prediction QC
+# 
+# In general, we've found that fitting survival models from scikit-survival to high-dimensional -omics data has been fairly error-prone. In particular, our models will often fail to converge: either there will be no valid hyperparameter combination in the search path that results in non-zero/NaN coefficients, or the predictor from the train set will make nonsensical predictions on the test data.
```
[the remark this answers is not in the document]
Comment (jjc2718) [reply 06_predict_survival/nbconverted/survival_results_qc.py:6]: Yes, the values are time from diagnosis until death. And yes, it's correct that this data is not available for many patients - these are called "censored" data points, and survival modeling techniques are designed for this scenario.

I don't think this is necessarily the reason that the models aren't converging, though; I think it's more likely to be a result of the way the optimization works for these models on high-dimensional data (although the number of censored samples may contribute). [This GitHub issue](https://github.com/sebp/scikit-survival/issues/28) describes some of the same optimization errors I've been running into, and the answer from the package maintainer was (paraphrased) "sometimes the optimization breaks, especially if you have lots of features, trying a larger hyperparameter range might help". I'm not sure exactly why this is more common for survival models than it is for classification or regression, but that's what I've observed.

File: 06_predict_survival/nbconverted/survival_results_qc.py [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+
+# In[5]:
+
+
+n_dim = 10
+
+sns.set({'figure.figsize': (24, 10)})
+fig, axarr = plt.subplots(2, 1)
+
+sns.pointplot(data=count_df[(count_df.signal == 'signal') & (count_df.n_dim == n_dim)],
```
[the remark this answers is not in the document]
Comment (jjc2718) [reply 06_predict_survival/nbconverted/survival_results_qc.py:97]: Those are both pretty sparsely sampled cancer types (CHOL has 45 samples and DLBC has 48). When the dataset is that small, my guess is that model convergence is pretty unstable and depends a lot on things like the dataset split and starting conditions. 

We're not using the models trained on these cancer types in the paper anyway, so I'm not too concerned.

File: 06_predict_survival/nbconverted/survival_results_qc.py [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+count_df.reset_index(inplace=True)
+count_df.head(10)
+
+
+# In[8]:
+
+
+n_dim = 10
+
+sns.set({'figure.figsize': (24, 10)})
```
[the remark this answers is not in the document]
Comment (jjc2718) [reply 06_predict_survival/nbconverted/survival_results_qc.py:159]: Yep, exactly!

File: 06_predict_survival/nbconverted/plot_survival_results.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjc2718) [reply 06_predict_survival/nbconverted/plot_survival_results.py:79]: The models trained on -omics data also include the clinical metadata (age, mutation burden, cancer type). So we would expect/hope that when we add -omics data we get better predictions on the held-out samples than we do without the -omics info. In general, this does seem to be the case.

File: 06_predict_survival/nbconverted/survival_results_qc.py [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+plt.title('Number of failed fits per cancer type')
+
+
+# ### Count results files, expression vs. methylation baseline
+
+# In[15]:
+
+
+# get baseline predictor results, using non-omics covariates only
+me_baseline_results_df = su.load_survival_results(me_baseline_results_dir)
```
[the remark this answers is not in the document]
Comment (jjc2718) [reply 06_predict_survival/nbconverted/survival_results_qc.py:255]: The baseline uses the same samples as the non-baseline models, just a subset of the features.
